compiler/src/package/schema.rs

```diff
@@ -0,0 +1,45 @@
+use super::{Error, Result};
+use helper::io::{self, Error as IOError};
+use serde_json::Value;
+use std::path::PathBuf;
+use url::Url;
+use valico::json_schema::Scope;
+
+pub struct Schema {
+	scope: Scope,
+	id: Url,
+}
+
+impl Schema {
+	pub fn new(schema: Value) -> Result<Self> {
+		let mut scope = Scope::new();
+		let id = scope
+			.compile(schema, false)
+			.map_err(|_err| Error::InvalidSchema)?;
+
+		Ok(Schema { scope, id })
+	}
+
+	pub fn from_path(path: PathBuf) -> Result<Self> {
+		match io::from_file(&path) {
+			Ok(contents) => {
+				let schema: Value =
+					serde_json::from_str(&contents).map_err(|_err| Error::InvalidSchema)?;
+
+				let schema = Self::new(schema)?;
+
+				Ok(schema)
+			}
+			Err(IOError::NotFound) => Err(Error::NoSchema),
+			_ => Err(Error::InvalidSchema),
+		}
+	}
+}
+
+impl Schema {
```

Comment:
I don't know much `Rust`, but why have you created another `impl` instead of putting inside the above `impl`??

Reply:
Because there's no explicit annotation of whether a function is a method or an associated function ("static" methods), by separating them, we don't need to look at the parameters to distinguish between the types.

Maybe I should've added a documentation comment to make the distinction clearer.

Reply:
If it's common in `Rust`you shouldn't

Reply:
It's common to divide, but there are no standard criteria, [`jrsonnet`](https://github.com/CertainLach/jrsonnet/blob/master/crates/jrsonnet-evaluator/src/lib.rs#L362-L462) splits the `impl`s based on their responsibility on the rendering process. I'll add the documentation comments like what is done on `jrsonnet` since I followed some of its patterns in other places.